bindings/src/index.ts

```diff
@@ -119,10 +127,15 @@ class NEARBindingsBuilder extends BaseVisitor {
     var signature = node.signature;
     let name = toString(node.name);
     var args: SimpleField[] = signature.parameters.map(param => {
-      return { name: toString(param.name), type: param.type };
+      let name = toString(param.name);
+      let type = param.type;
+      return { name,
+        type,
+        isGeneric: node.typeParameters != null && node.typeParameters.some(T => toString(T) == toString(type))
```

Comment:
What does the second condition mean?

Reply:
Is there a type parameter that equals the type of the field.  This was the easiest way I could to determine if a type is a generic with the AST.  I think perhaps we need to a postCompile phase that can do a better job at type resolution.

Reply:
Okay it feels somewhat hacky but fine for the moment

Reply:
Yeah, but the good thing about my current strategy is that we only have to determine if it's generic and then it's a runtime check to determine if the correct type was decoded.